src/NLog/Config/XmlLoggingConfiguration.cs

```diff
@@ -272,7 +272,7 @@ public override IEnumerable<string> FileNamesToWatch
         public override LoggingConfiguration Reload()
         {
             if (!string.IsNullOrEmpty(_originalFileName))
-                return new XmlLoggingConfiguration(_originalFileName, LogFactory);
+                return LogFactory.CreateConfig(_originalFileName);
```

Comment:
Still curious why this change is needed?

Reply:
Because the test still fails, reload will search on the disk and fails

Reply:
Even when you merge/rebase with master, and take in #3941 ?

Reply:
yes I did a rebase.

Reply:
See https://github.com/NLog/NLog/compare/logFactory.ReloadConfiguration...master

Reply:
Really strange that it tries to access the disk when it can only access the disk using the mocked AppEnvironment. Are you mocking AppEnvironment ?

Reply:
I think the mocking is correct.

See branch loadconfiguration-applyOnReload_noLogFactory.CreateConfig

![image](https://user-images.githubusercontent.com/5808377/81506828-62c7e180-92f9-11ea-965c-b7f5224f6c35.png)

Reply:
I see the issue, LogFactory needs IAppEnvironment, while it's already in ILoggingConfigurationLoader

Reply:
Fixed